src/modules/exec.ts

```diff
@@ -33,25 +33,30 @@ export function setSpawnImplementation(func: (command: string, args: string | st
     ptySpawn = func;
 }
 
-var ptySpawn = (command: string, args: string | string[], options: any): IPty => spawn(command, args, options);
+let ptySpawn = (command: string, args: string | string[], options: any): IPty => spawn(command, args, options);
 
 export async function awaitSpawn(
     command: string,
     args: string[],
     cwd: string,
     env: NodeJS.ProcessEnv,
     writeStdout: boolean,
+    interactive: boolean = false,
 ): Promise<{ exitCode: number; signal?: number } | null> {
     const projectCache = ProjectCache.read();
 
-    var ptyProcess = ptySpawn(command, args, {
+    const ptyProcess = ptySpawn(command, args, {
         cwd: cwd,
         env: env as any,
     });
 
     ptyProcess.onData((data) => lineCapturer(projectCache, writeStdout, data));
 
-    process.stdin.on('data', ptyProcess.write.bind(ptyProcess));
+    if (interactive) {
+        process.stdin.setRawMode(true);
+    }
+
+    process.stdin.on('data', (data) => ptyProcess.write(data.toString()));
```

Comment:
Not sure if it is good to call `toString()` on the data which we should keep in its data form to account for special character sequences to be passed.